### PR TITLE
specific patches

### LINKED ISSUE #43: Recreate various specific area patches
Too many to list at this moment

### PR DESCRIPTION
**Pull request by duncathan**
unfinished and untested

also this probably crashes due to missing pak dependencies until #49 is merged and working

fixes #43

### CODE REVIEW

File: src/open_prime_rando/echoes/specific_area_patches/required_fixes.py [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+    area = editor.get_area(AGON_WASTES_MLVL, agon_wastes.MAIN_REACTOR_MREA)
+
+    unload_relay = area.get_instance("Unload dock when door closed")
+    trigger = area.get_instance("Trigger Start DS Intro")
+    trigger.add_connection(State.Entered, Message.SetToZero, unload_relay)
+
+
+def sacrificial_chamber(editor: PatcherEditor):
+    """
+    Unknown purpose.
```
Comment (henriquegemignani) [inline src/open_prime_rando/echoes/specific_area_patches/required_fixes.py:80]: I remember there being a bug where the pickup only appears the first time you trigger the fight.
So you could enter the room, trigger the boss, leave, come back again, kill the boss this time and get a big nothing.

Comment (duncathan) [reply src/open_prime_rando/echoes/specific_area_patches/required_fixes.py:80]: that reminds mii - it's still possible to remove the alpha splinter pickup permanently, isn't it?

Comment (henriquegemignani) [reply src/open_prime_rando/echoes/specific_area_patches/required_fixes.py:80]: I'd say yes. Trigger fight, jump over walls and leave.

File: src/open_prime_rando/echoes/specific_area_patches/required_fixes.py [hunk 2]
```diff
@@ -123,6 +129,37 @@ def main_research(editor: PatcherEditor):
...
+    spawn_xfm = portal_spawn.get_properties_as(SpawnPoint).editor_properties.transform
+
+    contraption_controller = area.get_layer("Default").add_instance_with(ScriptLayerController(
+        editor_properties=EditorProperties(
+            name="Dynamic Increment Contraption",
+            transform=spawn_xfm
+        ),
+        layer=LayerSwitch(
+            area_id=sanctuary_fortress.MAIN_RESEARCH_INTERNAL_ID,
+            layer_number=6
```
Comment (henriquegemignani) [inline src/open_prime_rando/echoes/specific_area_patches/required_fixes.py:143]: Could use `.get_layer("...").index` so we have numbers :)

Comment (duncathan) [reply src/open_prime_rando/echoes/specific_area_patches/required_fixes.py:143]: oh yea true

File: src/open_prime_rando/echoes/specific_area_patches/required_fixes.py (hunk 1 above)
Comment (dyceron) [reply src/open_prime_rando/echoes/specific_area_patches/required_fixes.py]: Yes to both Grapple Guardian and Alpha Splinter

File: src/open_prime_rando/echoes/specific_area_patches/required_fixes.py [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+    SANCTUARY_FORTRESS_MLVL,
+    TEMPLE_GROUNDS_MLVL,
+    TORVUS_BOG_MLVL,
+)
+from open_prime_rando.patcher_editor import PatcherEditor
+
+LOG = logging.getLogger("echoes_patcher")
+
+
+def required_fixes(editor: PatcherEditor):
```
Comment (dyceron) [inline src/open_prime_rando/echoes/specific_area_patches/required_fixes.py:34]: If you reload Undertemple without collecting the pickup from PB Guardian, it is lost forever.

Comment (dyceron) [reply src/open_prime_rando/echoes/specific_area_patches/required_fixes.py:34]: Related, add a HUD memo for Dark Oasis regarding PB Guardian.

Comment (dyceron) [inline src/open_prime_rando/echoes/specific_area_patches/required_fixes.py:34]: The scan for activating the cannon in Sacred Bridge is deactivated after lowering the gate (and I think reloading the room). Keeping the scan always active will prevent players from softlocking down there if they have no items to climb back up and the cannon wasn't activated prior.

File: src/open_prime_rando/echoes/custom_assets.py [hunk 4]
```diff
@@ -120,6 +122,18 @@ def _create_split_ammo(editor: AssetManager):
         editor.add_new_asset(f"{ammo.name}_ancs", ancs, editor.find_paks(BEAM_AMMO_EXPANSION_ANCS))
 
 
+def _import_premade_assets(editor: AssetManager):
+    assets = Path(__file__).parent.joinpath("custom_assets", "general")
+    for f in assets.glob("*"):
```
Comment (henriquegemignani) [inline src/open_prime_rando/echoes/custom_assets.py:127]: `.glob("*.*")`, and then remove the is_dir check?

**Review by henriquegemignani — APPROVED** [review]

**Pull request merged**